### PR TITLE
Changes to pokedex functionality and Solution to Issue#185

### PR DESCRIPTION
**Pull request by Ellamental2**
added functionality to the pokédex macro so if the user would not normally have permission to view the pokemon's pokédex page, they instead get a summary of the pokemon, and the pokemons dex entry is added to the users character sheet under "seen".

example: https://youtu.be/38kTVUpvb3Q

### CODE REVIEW

**Comment by Ellamental2** [comment]
This could also be given an additional settings of.
"Automatically add pokemon to player's dex page"
and
"Show players basic info when not allowed to see full details"
and wrap the lines in the ptu.js in ifs

**Comment by Ellamental2** [comment]
added solution to issue https://github.com/dylanpiera/Foundry-Pokemon-Tabletop-United-System/issues/185

added option to obscure move effectiveness in sidebar. this is in PTU Settings General 
image


options for this are Disabled - no-one can see move effectiveness
GM Only - only GM can see move effectiveness
All Players - all players can see move effectiveness.

###Examples (GM View on Left, Player view on right)

**Disabled**
image

**GM Only**
image

**All Players**
image

**Comment by Ellamental2** [comment]
Added Extended Pokédex options so GMs can choose whether the pokemon are automatically added to the players dex, and when full details will be shown.
plus re-implemented a completely Disabled option

### New Options
image

image

**Comment by Ellamental2** [comment]
Reordered the options for Pokédex Permissions to make more sense
image

**Comment by Ellamental2** [comment]
added options in move effectiveness to only show on seen/owned pokemon

image

**Comment by Ellamental2** [comment]
TODO:

image

- [ ] Comment out line 770, uncomment 771

File: module/sidebar/components/moves-component.js [hunk 1]
```diff
@@ -58,13 +59,73 @@ export default class MovesList extends Component {
...
+                            moveData.data.effectiveness = -1;
+                            break;
+                        }
+
+                        if (game.user.targets.size < 1) {
+                            moveData.data.effectiveness = -1;
+                            break;
+                        }
+
+                        for (let token of game.user.targets.size > 0 ? game.user.targets.values() : canvas.tokens.controlled) {
```
Comment (dylanpiera) [inline module/sidebar/components/moves-component.js:89]: Is this for loop necessary? If multiple-mons are selected effectiveness is always set to x1

File: module/sidebar/components/moves-component.js [hunk 2]
```diff
@@ -58,13 +59,73 @@ export default class MovesList extends Component {
...
+                            ui.notifications.warn("Please make sure you have a trainer as your Selected Player Character");
+                            moveData.data.effectiveness = -1;
+                            break;
+                        }
+                        
+                        if (game.user.targets.size < 1) {
+                            moveData.data.effectiveness = -1;
+                        }
+
+                        for (let token of game.user.targets.size > 0 ? game.user.targets.values() : canvas.tokens.controlled) {
```
Comment (dylanpiera) [inline module/sidebar/components/moves-component.js:110]: Is this for loop necessary? If multiple-mons are selected effectiveness is always set to x1

**Review by dylanpiera — CHANGES_REQUESTED** [review]
See my earlier comment, if you can test and fix that I'll merge it with the next changes (I'm off for now)

File: module/sidebar/components/moves-component.js (hunk 1 above)
Comment (Ellamental2) [reply module/sidebar/components/moves-component.js]: unnecessary loops removed

File: module/sidebar/components/moves-component.js (hunk 2 above)
Comment (Ellamental2) [reply module/sidebar/components/moves-component.js]: unnecessary loops removed

**Pull request merged**